core/handler/manager_server.go

```diff
@@ -291,8 +293,8 @@ func (h *handlerManager) GetApplication(ctx context.Context, in *pb.ApplicationI
 }
 
 func (h *handlerManager) RegisterApplication(ctx context.Context, in *pb.ApplicationIdentifier) (*empty.Empty, error) {
-	if !in.Validate() {
-		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Application Identifier", "validation failed"))
+	if err := in.Validate(); err != nil {
+		return nil, errors.BuildGRPCError(errors.NewErrInvalidArgument("Application Identifier", err.Error()))
```

Comment:
Be consistent with `errors.Wrap` vs wrapping it in `errors.NewErrInvalidArgument`